src/utils/manipulable-promise.ts

```diff
@@ -0,0 +1,32 @@
+import { Promise } from 'es6-promise';
+
+export interface ManipulablePromise<T> extends Promise<T> {
```

Comment:
My understanding from the code is that `ManipulablePromise` exists to give talker access to the resolve and reject calls; is that right? Could you add a brief comment explaining that?